autoloop: include peer rules for inflight/budget limits

carlaKC opened the pull request:
If we ran into a constraint while dispatching new autoloops (hit our in-flight limit or budget), we were not setting a disqualified reason, which would mean that this reasoning isn't surfaced to the user. This PR adds a test to demonstrate the bug and addresses it.

#### Pull Request Checklist
- [x] Update `release_notes.md` if your PR contains major features, breaking changes or bugfixes

bhandras commented on liquidity/interface.go:
```diff
@@ -69,3 +75,24 @@ func (l *loopOutSwapSuggestion) channels() []lnwire.ShortChannelID {
+
+// peers returns the set of peers that the loop out swap is restricted to.
+func (l *loopOutSwapSuggestion) peers(knownChans map[uint64]route.Vertex) []route.Vertex {
+	peers := make(map[route.Vertex]struct{})
```
nit: `make(map[route.Vertex]struct{}, len(l.OutgoingChanSet))`

bhandras commented on liquidity/interface.go:
```diff
@@ -69,3 +75,24 @@ func (l *loopOutSwapSuggestion) channels() []lnwire.ShortChannelID {
 }
+
+// peers returns the set of peers that the loop out swap is restricted to.
+func (l *loopOutSwapSuggestion) peers(knownChans map[uint64]route.Vertex) []route.Vertex {
```
nit: could break the line

bhandras reviewed (approved):
LGTM :palm_tree:

guggero reviewed (approved):
LGTM :tada:

carlaKC commented:
Addressed nits and added a compile time assertion that we satisfy our interface - [diff](https://github.com/lightninglabs/loop/compare/d265bc9bee654b0f69c986001e2d7d323e4c33ed..cc10d5b207568f5d8b0dabd5bb5e61913f38ac1a).
